siddhi-test-framework/src/main/java/io/siddhi/distribution/test/framework/SiddhiRunnerContainer.java

```diff
@@ -209,18 +214,43 @@ private int getStartupTimeoutSeconds() {
     @Override
     protected void waitUntilContainerStarted() {
         logger().info("Waiting for Siddhi Runner Container to start...");
-        retryUntilSuccess(getStartupTimeoutSeconds(), TimeUnit.SECONDS, () -> {
-            if (!isRunning()) {
-                throw new ContainerLaunchException("Siddhi Runner Container failed to start");
-            }
-            HTTPClient.HTTPResponseMessage httpResponseMessage = callHealthAPI();
-            if (httpResponseMessage.getResponseCode() == 200) {
-                logger().info("Siddhi Runner Health API reached successfully.");
-                return null;
-            } else {
-                throw new ConnectException("Failed to connect with the Siddhi Runner health API");
+        if (deploymentDirectoryFile != null) {
+            String[] siddhiAppsArray = deploymentDirectoryFile.list();
+            assert siddhiAppsArray != null;
+            for (String siddhiApp : siddhiAppsArray) {
+                String fileName = siddhiApp.substring(0, siddhiApp.length() - 7);
+                retryUntilSuccess(getStartupTimeoutSeconds(), TimeUnit.SECONDS, () -> {
+                    if (!isRunning()) {
+                        throw new ContainerLaunchException("Siddhi Runner Container failed to start");
+                    }
+                    HTTPClient.HTTPResponseMessage httpResponseMessage = callHealthAPI();
+                    if (httpResponseMessage.getResponseCode() == 200) {
+                        String logs = this.getLogs();
+                        if (logs.contains("Siddhi App " + fileName + " deployed successfully")) {
+                            logger().info("Siddhi Runner Health API reached successfully.");
+                        } else {
+                            throw new Exception("Siddhi App AppDeploymentTestResource deployment failed.");
+                        }
+                        return null;
+                    } else {
+                        throw new ConnectException("Failed to connect with the Siddhi Runner health API");
+                    }
+                });
             }
-        });
+        } else {
```

Comment:
Why are we having duplicated code

Reply:
Fixed in https://github.com/siddhi-io/distribution/commit/684ff2789b8193d4b3820859614a6de441dbe914